include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -127,6 +127,15 @@ struct all_view_fn
     {
         return ::std::forward<_R>(__r);
     }
+#if _ONEDPL_CPP20_RANGES_PRESENT
+    template <typename _T>
+    auto
+    operator()(std::vector<_T, sycl::usm_allocator<_T, sycl::usm::alloc::shared>>& __vec) const 
```

Comment:
```suggestion
    template <typename _T, whatever_enum_name _MemoryKind>
    auto
    operator()(std::vector<_T, sycl::usm_allocator<_T, _MemoryKind>>& __vec) const 
```

I think it's better to support any kind of memory, not just `sycl::usm::alloc::shared`

Reply:
done as    
```
    template <typename _T, sycl::usm::alloc _AllocType>
    auto
    operator()(std::vector<_T, sycl::usm_allocator<_T, _AllocType>>& __vec) const 

```